clouddriver-appengine/src/main/groovy/com/netflix/spinnaker/clouddriver/appengine/provider/agent/AppEngineLoadBalancerCachingAgent.groovy

```diff
@@ -0,0 +1,237 @@
+import static com.netflix.spinnaker.clouddriver.appengine.cache.Keys.Namespace.ON_DEMAND
+
+@Slf4j
+class AppEngineLoadBalancerCachingAgent extends AbstractAppEngineCachingAgent implements OnDemandAgent {
+  final String category = "loadBalancer"
+
+  final OnDemandMetricsSupport metricsSupport
+
+  static final Set<AgentDataType> types = Collections.unmodifiableSet([
+    AUTHORITATIVE.forType(LOAD_BALANCERS.ns),
+  ] as Set)
+
+  String agentType = "${accountName}/${AppEngineLoadBalancerCachingAgent.simpleName}"
+
+  @Override
+  String getSimpleName() {
+    AppEngineLoadBalancerCachingAgent.simpleName
+  }
+
+  @Override
+  String getOnDemandAgentType() {
+    "${agentType}-OnDemand"
+  }
+
+  @Override
+  Collection<AgentDataType> getProvidedDataTypes() {
+    types
+  }
+
+  AppEngineLoadBalancerCachingAgent(String accountName,
+                                    AppEngineNamedAccountCredentials credentials,
+                                    ObjectMapper objectMapper,
+                                    Registry registry) {
+    super(accountName, objectMapper, credentials)
+    metricsSupport = new OnDemandMetricsSupport(
+      registry,
+      this,
+      "$AppEngineCloudProvider.ID:$OnDemandAgent.OnDemandType.LoadBalancer")
+  }
+
+  @Override
+  boolean handles(OnDemandAgent.OnDemandType type, String cloudProvider) {
+    type == OnDemandAgent.OnDemandType.LoadBalancer && cloudProvider == AppEngineCloudProvider.ID
+  }
+
+  @Override
+  OnDemandResult handle(ProviderCache providerCache, Map<String, ? extends Object> data) {
+    if (!data.containsKey("loadBalancerName") || data.account != accountName) {
+      return null
+    }
+
+    def loadBalancerName = data.loadBalancerName.toString()
+
+    Service service = metricsSupport.readData {
+      loadService(loadBalancerName)
+    }
+
+    CacheResult result = metricsSupport.transformData {
+      buildCacheResult([service], [:], [], Long.MAX_VALUE)
+    }
+
+    def jsonResult = objectMapper.writeValueAsString(result.cacheResults)
+    def loadBalancerKey = Keys.getLoadBalancerKey(accountName, loadBalancerName)
+    if (result.cacheResults.values().flatten().isEmpty()) {
+      providerCache.evictDeletedItems(ON_DEMAND.ns, [loadBalancerKey])
+    } else {
+      metricsSupport.onDemandStore {
+        def cacheData = new DefaultCacheData(
+          loadBalancerKey,
+          10 * 60, // ttl is 60 minutes.
+          [
+            cacheTime: System.currentTimeMillis(),
+            cacheResults: jsonResult,
+            processedCount: 0,
+            processedTime: 0
+          ],
+          [:]
+        )
+
+        providerCache.putCacheData(ON_DEMAND.ns, cacheData)
+      }
+    }
+
+    Map<String, Collection<String>> evictions = service ? [:] : [(LOAD_BALANCERS.ns): [loadBalancerKey]]
+
+    log.info "On demand cache refresh (data: ${data}) succeeded."
+
+    new OnDemandAgent.OnDemandResult(
```

Comment:
Let's use explicit `return` statements for methods that aren't simple one-liners.